hack/gcp-service-account-maintenance/build-service-account.bash

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+# Confirm execution from correct directory
+if [ ! -f ./config.source ]; then
+    echo "Not in tool root";
+    exit 1;
+fi
+source ./config.source
+
+DATE=`date +%Y%m%d-%H%M%S`
+
+# Used in referencing for delete/enable commands
+GCLOUD_SA_EMAIL="${GCLOUD_SA}@${GCLOUD_PROJECT}.iam.gserviceaccount.com"
+
+GCLOUD="gcloud --project ${GCLOUD_PROJECT}"
+
+echo
+echo "checking connection:"
+CURRENT_OC_CONTEXT=`oc config view -o json | jq -r '.["current-context"]'`
+CURRENT_OC_SERVER=`oc config view -o json | jq -r ".contexts[] | select(.name == \"${CURRENT_OC_CONTEXT}\") | .context.cluster"`
+if [ ! "${CURRENT_OC_SERVER}" == "${TARGET_OC_SERVER}" ]; then
+    echo "oc cli configured to incorrect context, expected [${TARGET_OC_SERVER}] got [${CURRENT_OC_SERVER}]";
+    echo "please log into your expected cluster and try again, or reconfigure ./config.source";
+    exit 1;
+fi
+echo "connected to [${CURRENT_OC_SERVER}]"
+
+echo
+echo "checking ${GCLOUD}:"
+GCLOUD_USER=`${GCLOUD} config get-value core/account`
+if [ ! "${GCLOUD_USER}" == "${TARGET_GCLOUD_USER}" ]; then
+    echo "oc cli configured to incorrect context, expected [${TARGET_GCLOUD_USER}] got [${GCLOUD_USER}]";
+    echo "please log into your expected gcloud account and try again, or reconfigure ./config.source";
+    exit 1;
+fi
+echo "connected to ${GCLOUD} with [${GCLOUD_USER}]"
+
+
+echo
+echo "checking existing service account [${GCLOUD_SA}]"
+${GCLOUD} iam service-accounts list --quiet --filter name:"${GCLOUD_SA}" | grep "${GCLOUD_SA}"
+[ $? == 0 ] && { SA_EXISTS=1; } || SA_EXISTS=0
```

Comment:
I'd prefer to check the return code of the command right in the if for better readability.